src/uct/ib/ud/accel/ud_mlx5.c

```diff
@@ -823,28 +834,87 @@ static uct_iface_ops_t uct_ud_mlx5_iface_tl_ops = {
     .iface_is_reachable       = uct_ib_iface_is_reachable
 };
 
+static ucs_status_t
+uct_ud_mlx5dv_calc_tx_wqe_ratio(uct_ib_mlx5_md_t *md)
+{
+    uct_ib_qp_init_attr_t qp_init_attr = {};
+    uct_ib_device_t *dev               = &md->super.dev;
+    ucs_status_t status;
+    struct ibv_qp *qp;
+    uct_ib_mlx5dv_qp_init_objs_t qp_init_objs;
+
+    status = uct_ib_mlx5dv_qp_init_objs_create(&md->super, &qp_init_objs);
+    if (status != UCS_OK) {
+        goto out;
+    }
+
+    uct_ib_mlx5dv_qp_init_attr(&qp_init_attr, md->super.pd, &qp_init_objs,
+                               IBV_QPT_UD, 128);
+    uct_ud_mlx5_qp_update_caps(&qp_init_attr.cap);
+
+#if HAVE_DECL_IBV_CREATE_QP_EX
+    qp = UCS_PROFILE_CALL_ALWAYS(ibv_create_qp_ex, dev->ibv_context,
+                                 &qp_init_attr);
+#else
+    qp = UCS_PROFILE_CALL_ALWAYS(ibv_create_qp, pd, &qp_init_attr);
```

Comment:
profiling on non-data path

Reply:
it was added before my changes.
let's discuss and remove it in another PR